crates/omnix-cli/src/command/develop.rs

```diff
@@ -28,7 +28,11 @@ impl DevelopCommand {
     pub async fn run(&self) -> anyhow::Result<()> {
         let flake = self.flake_shell.without_attr();
 
-        let om_config = OmConfig::from_flake_url(NixCmd::get().await, &flake).await?;
+        let nix_cmd = NixCmd::get().await;
+        let om_config = match OmConfig::from_yaml(nix_cmd, &flake).await {
+            Err(OmConfigError::ReadYaml(_)) => OmConfig::from_flake(nix_cmd, &flake).await,
+            other => other,
+        }?;
```

Comment:
I see this logic being repeated in 3 places; does it make sense to encapsulate away in `OmConfig`?

Also, the catch `ReadYaml` error fallback means legitimate syntax issues in `om.yaml` will erroneously fallback to evaluating to Nix (when the correct behaviour is to crash the omnix process with the appropriate error). Falling back should happen only when `om.yaml` is missing.

Reply:
https://github.com/juspay/omnix/pull/352/commits/1a4916efe435269dbd919c823b38ca17eb878ddc